tests/test_skyWcs.py

```diff
@@ -0,0 +1,123 @@
+    """
+
+    def setUp(self):
+        TransformTestBaseClass.setUp(self)
+        self.crpix = Point2D(100, 100)
+        self.crval = SpherePoint(30 * degrees, 45 * degrees)
+        self.scale = 1.0 * arcseconds
+
+    def checkTanWcs(self, orientation, flipX):
+        """Construct a pure TAN SkyWcs and check that it operates as specified
+
+        Parameters
+        ----------
+        orientation : lsst.afw.geom.Angle
+            Position angle of focal plane +Y, measured from N through E.
+            At 0 degrees, +Y is along N and +X is along E/W if flipX false/true
+            At 90 degrees, +Y is along E and +X is along S/N if flipX false/true
+        flipX : bool
+            Fip x axis? See `orientation` for details.
+
+        Returns
+        -------
+        wcs : SkyWcs
+            The generated pure TAN SkyWcs
+        """
+        wcs = SkyWcs(
+            crpix = self.crpix,
+            crval = self.crval,
+            cdMatrix = makeCdMatrix(scale=self.scale, orientation=orientation, flipX=flipX),
+        )
+        self.checkPersistence(wcs)
+
+        xoffAng = 180*degrees if flipX else 0*degrees
+
+        pixelList = [
+            Point2D(self.crpix[0], self.crpix[1]),
+            Point2D(self.crpix[0] + 1, self.crpix[1]),
+            Point2D(self.crpix[0], self.crpix[1] + 1),
+        ]
+        skyList = wcs.tranForward(pixelList)
+
+        # check pixels to sky
+        predSkyList = [
+            self.crval,
+            self.crval.offset(xoffAng - orientation, self.scale),
+            self.crval.offset(90*degrees - orientation, self.scale)
+        ]
+        self.assertSpherePointListsAlmostEqual(predSkyList, skyList)
+        self.assertSpherePointListsAlmostEqual(predSkyList, wcs.pixelToSky(pixelList))
+        for pixel, predSky in zip(pixelList, predSkyList):
+            self.assertSpherePointsAlmostEqual(predSky, wcs.pixelToSky(pixel))
+            anglePair = wcs.pixelToSky(*pixel)
+            self.assertSpherePointsAlmostEqual(predSky, SpherePoint(*anglePair))
+
+        # check sky to pixels
+        self.assertPairListsAlmostEqual(pixelList, wcs.tranInverse(skyList))
+        self.assertPairListsAlmostEqual(pixelList, wcs.tranInverse(skyList))
+        for pixel, sky in zip(pixelList, skyList):
+            self.assertPairsAlmostEqual(pixel, wcs.skyToPixel(sky))
+            xyPair = wcs.skyToPixel(*sky)
+            self.assertPairsAlmostEqual(pixel, Point2D(*xyPair))
+
+        crval = wcs.getSkyOrigin()
+        self.assertSpherePointsAlmostEqual(crval, self.crval)
+
+        crpix = wcs.getPixelOrigin()
+        self.assertPairsAlmostEqual(crpix, self.crpix)
+
+        cdMatrix = wcs.getCdMatrix()
+        predCdMatrix = makeCdMatrix(scale=self.scale, orientation=orientation, flipX=flipX)
+        self.assertFloatsAlmostEqual(cdMatrix, predCdMatrix)
+
+        pixelScale = wcs.getPixelScale(self.crpix)
+        self.assertAnglesAlmostEqual(self.scale, pixelScale)
```

Comment:
Shouldn't they be exact?  At least set the allowed difference down to machine precision rather than leaving the default.